pkg/artifactory/resource/repository/remote/resource_artifactory_remote_docker_repository.go

```diff
@@ -8,58 +8,63 @@ import (
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Computed:    true,
+				Description: "When set, Artifactory will block the pulling of Docker images with manifest v2 schema 1 from the remote repository (i.e. the upstream). It will be possible to pull images with manifest v2 schema 1 that exist in the cache.",
+			},
+			// We need to set default to ["**"] once we migrate to plugin-framework. SDKv2 doesn't support that.
+			"external_dependencies_patterns": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				RequiredWith: []string{"external_dependencies_enabled"},
+				Description: "An allow list of Ant-style path patterns that determine which remote VCS roots Artifactory will " +
+					"follow to download remote modules from, when presented with 'go-import' meta tags in the remote repository response. " +
+					"By default, this is set to '**' in the UI, which means that remote modules may be downloaded from any external VCS source." +
+					"Due to SDKv2 limitations, we can't set the default value for the list." +
+					"This value must be assigned to the attribute manually, if user don't specify any other non-default values." +
+					"This attribute must be set together with `external_dependencies_enabled = true`",
 			},
-			RequiredWith: []string{"external_dependencies_enabled"},
-			Description: "An allow list of Ant-style path patterns that determine which remote VCS roots Artifactory will " +
-				"follow to download remote modules from, when presented with 'go-import' meta tags in the remote repository response. " +
-				"By default, this is set to '[**]' in the UI, which means that remote modules may be downloaded from any external VCS source." +
-				"Due to SDKv2 limitations, we can't set the default value for the list." +
-				"This value [**] must be assigned to the attribute manually, if user don't specify any other non-default values." +
-				"We don't want to make this attribute required, but it must be set to avoid the state drift on update. Note: Artifactory assigns " +
-				"[**] on update if HCL doesn't have the attribute set or the list is empty.",
 		},
-	}, repository.RepoLayoutRefSchema("remote", packageType))
+		repository.RepoLayoutRefSchema(rclass, DockerPackageType),
+	)
+}
 
-	type DockerRemoteRepository struct {
-		RepositoryRemoteBaseParams
-		ExternalDependenciesEnabled  bool     `hcl:"external_dependencies_enabled" json:"externalDependenciesEnabled"`
-		ExternalDependenciesPatterns []string `hcl:"external_dependencies_patterns" json:"externalDependenciesPatterns"`
-		EnableTokenAuthentication    bool     `hcl:"enable_token_authentication" json:"enableTokenAuthentication"`
-		BlockPushingSchema1          bool     `hcl:"block_pushing_schema1" json:"blockPushingSchema1"`
-	}
+type DockerRemoteRepo struct {
+	RepositoryRemoteBaseParams
+	ExternalDependenciesEnabled  bool     `json:"externalDependenciesEnabled,omitempty"`
```

Comment:
Are you sure about adding `omitempty` for a boolean? I've removed it from the repository resource because it created an issue for the user, when they update the attribute.

Reply:
I think this is left over from before your fix. I'll remove this.